tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/semver/SemverValidatorTest.java

```diff
@@ -0,0 +1,129 @@
+    }
+
+    @Test(description = "Test semver validator output on a set of patch-incompatible changes")
+    public void testValidatePatchIncompatibleChanges() throws BallerinaTestException {
+        pushPackageToCustomRepo("package_1_0_0");
+
+        List<String> expectedLogs = new LinkedList<>();
+        expectedLogs.add("current version: 1.1.0");
+        expectedLogs.add("compared with the release version '1.0.0'");
+        expectedLogs.add("patch-incompatible changes detected");
+        expectedLogs.add("suggested version: 1.1.0");
+
+        executeSemverCommand("package_1_1_0", new String[]{"--compare-version=1.0.0"}, expectedLogs);
+    }
+
+    @Test(description = "Test semver validator output on a set of backward-compatible changes")
+    public void testValidateBackwardIncompatibleChanges() throws BallerinaTestException {
+        pushPackageToCustomRepo("package_1_0_0");
+
+        List<String> expectedLogs = new LinkedList<>();
+        expectedLogs.add("current version: 2.0.0");
+        expectedLogs.add("compared with the release version '1.0.0'");
+        expectedLogs.add("backward-incompatible changes detected");
+        expectedLogs.add("suggested version: 2.0.0");
+
+        executeSemverCommand("package_2_0_0", new String[]{"--compare-version=1.0.0"}, expectedLogs);
+    }
+
+    @Test(description = "Test semver validation functionality where multiple release versions of the same package" +
+            "are available in local/central repositories")
+    public void testValidateWithMultipleRelease() throws BallerinaTestException {
+        pushPackageToCustomRepo("package_1_0_0");
+        pushPackageToCustomRepo("package_1_0_1");
+        pushPackageToCustomRepo("package_1_1_0");
+
+        List<String> expectedLogs = new LinkedList<>();
+        expectedLogs.add("current version: 2.0.0");
+        expectedLogs.add("compared with the release version '1.0.1'");
+        expectedLogs.add("backward-incompatible changes detected");
+        expectedLogs.add("suggested version: 2.0.0");
+
+        executeSemverCommand("package_2_0_0", new String[]{"--compare-version=1.0.1"}, expectedLogs);
+    }
+
+    @Test(description = "Test semver validation output with the diff option enabled")
+    public void testValidateWithDiffEnabled() throws BallerinaTestException {
+        pushPackageToCustomRepo("package_1_0_0");
+
+        List<String> expectedLogs = new LinkedList<>();
+        expectedLogs.add("current version: 2.0.0");
+        expectedLogs.add("compared with the release version '1.0.0'");
+        expectedLogs.add("backward-incompatible changes detected");
+        expectedLogs.add("suggested version: 2.0.0");
+        // diff output related logs
+        expectedLogs.add("Comparing version '2.0.0'(local) with version '1.0.0'(central)");
+        expectedLogs.add("[+-] package 'library' is modified [version impact: MAJOR]");
+        expectedLogs.add("  [+-] module 'library' is modified [version impact: MAJOR]");
```

Comment:
I'm wonder whether there is a chance to change the white space, and tab space according to the OS and will it matter for test comparison?

Reply:
IMO we shouldn't be having issues as here we only have spaces. This test is passing in both windows and linux builds

Reply:
Then it would be fine :)